### PR TITLE
ARROW-6474: [Python] Add option to use legacy / pre-0.15 IPC message format and to set the default using PYARROW_LEGACY_IPC_FORMAT environment variable

### PR DESCRIPTION
**Pull request by wesm**
It feels gross to alter behavior with environment variables but this is probably the least invasive way to enable Apache Spark users to upgrade to pyarrow 0.15.0 or beyond if they are frozen on an older Spark release

### CODE REVIEW

**Comment by wesm** [comment]
cc @BryanCutler @pitrou if you have any nits about this

**Review by BryanCutler — APPROVED** [review]
LGTM, thanks @wesm !  I can work on updating future versions of Spark to not require this.

**Comment by nealrichardson** [comment]
Is it necessary for the env var to have PYARROW in the name (`PYARROW_LEGACY_IPC_FORMAT`)? Since we should probably do the same in R (https://issues.apache.org/jira/browse/ARROW-6539), would `ARROW_LEGACY_IPC_FORMAT` be better?

Moving onto the next word in the var name ;) "legacy" is always subjective/relative. We're always writing tomorrow's legacy code. Is there a more explicit word we can use, in case we have to feature-flag another "legacy" feature some time? Not trying to overthink this, just don't want us to underthink either.

**Comment by wesm** [comment]
We could call it "ARROW_PRE_0_15_IPC_FORMAT"

**Comment by nealrichardson** [comment]
WFM

File: python/pyarrow/ipc.py [hunk 1]
```diff
@@ -101,9 +105,21 @@ class RecordBatchFileWriter(lib._RecordBatchFileWriter):
         Either a file path, or a writable file object
     schema : pyarrow.Schema
         The Arrow schema for data to be written to the file
+    use_legacy_format : boolean, default None
+        If None, use True unless overridden by PYARROW_LEGACY_IPC_FORMAT=1
```
Comment (romainfrancois) [inline python/pyarrow/ipc.py]:
```suggestion
        If None, use True unless overridden by ARROW_PRE_0_15_IPC_FORMAT =1
```

File: python/pyarrow/ipc.py [hunk 2]
```diff
@@ -70,9 +70,13 @@ class RecordBatchStreamWriter(lib._RecordBatchStreamWriter):
...
```
Comment (romainfrancois) [inline python/pyarrow/ipc.py]:
```suggestion
        If None, use True unless overridden by ARROW_PRE_0_15_IPC_FORMAT =1
```

**Comment by wesm** [comment]
Think I fixed the sphinx warning. I'm going to let CI fully run just in case...

**Comment by wesm** [comment]
Travis CI: https://travis-ci.org/wesm/arrow/builds/586658822

**Pull request closed without merging**